Softeq.XToolkit.WhiteLabel.Droid/DroidBootstrapper.cs

```diff
@@ -46,31 +33,26 @@ protected override void RegisterInternalServices(IContainerBuilder builder)
             builder.PerDependency<DroidFrameNavigationService, IFrameNavigationService>(IfRegistered.Keep);
         }
 
-        private static Dictionary<Type, Type> CreateAndRegisterMissedViewModels(
-            IContainerBuilder builder,
-            IEnumerable<Assembly> assemblies)
+        protected override void RegisterViewLocator(IContainerBuilder builder, Dictionary<Type, Type> viewModelToViewMapping)
         {
-            var viewModelToViewTypes = new Dictionary<Type, Type>();
-            var targetTypes = assemblies
-                .SelectMany(assembly => assembly.GetTypes()
-                    .View(typeof(AppCompatActivity), typeof(Fragment), typeof(DialogFragment)));
-
-            foreach (var type in targetTypes)
+            builder.Singleton<IViewLocator>(container =>
             {
-                var viewModelType = type.BaseType.GetGenericArguments()[0];
-
-                if (!viewModelType.IsAbstract)
-                {
-                    viewModelToViewTypes.Add(viewModelType, type);
-                    builder.PerDependency(viewModelType, IfRegistered.Keep);
-                }
-            }
-
-            return viewModelToViewTypes;
+                var viewLocator = container.Resolve<DroidViewLocator>();
```

Comment:
Does our IoC-container support registration with parameters? If so, it would be much better to just register DroidViewLocator as IViewLocator, providing viewModelToViewMapping as custom parameter.
Current implementation seems confusing, as it requires specific type to be already registered in IoC-container beforehand

Reply:
Great idea! Done.